src/napari_imagej/_function.py

```diff
@@ -309,37 +406,69 @@ def show_tabular_output():
 
 
 def _add_napari_metadata(execute_module: Callable, info, unresolved_inputs):
-    menu_string = " > ".join(str(p) for p in info.getMenuPath())
-    execute_module.__doc__ = f"Invoke ImageJ2's {menu_string} command"
-    execute_module.__name__ = re.sub("[^a-zA-Z0-9_]", "_", menu_string)
-    execute_module.__qualname__ = menu_string
+    module_name = ij.py.from_java(info.getTitle())
+    execute_module.__doc__ = f"Invoke ImageJ2's {module_name}"
+    execute_module.__name__ = module_name
+    execute_module.__qualname__ = module_name
 
     # Rewrite the function signature to match the module inputs.
     _modify_function_signature(execute_module, unresolved_inputs, info)
 
     # Add the type hints as annotations metadata as well.
     # Without this, magicgui doesn't pick up on the types.
     type_hints = {
-        str(i.getName()): _ptype(i.getType()) for i in unresolved_inputs
+        str(i.getName()): _ptype(i) for i in unresolved_inputs
     }
-    out_types = [o.getType() for o in info.outputs()]
-    return_annotation = _ptype(out_types[0]) if len(out_types) == 1 else dict
+    return_annotation = _ptype(info.outputs()[0]) if len(info.outputs()) == 1 else dict
```

Comment:
Why is the default a `dict`?

Reply:
This is just how you do it.

Reply:
Very informative @gselzer :confused:

I made a separate issue for this, as I think it is actually wrong. If the `Module` returned multiple outputs via `ModuleItem`s, then we should return both to napari. See https://github.com/imagej/napari-imagej/issues/20